vm/instruction_translator.go

```diff
@@ -44,87 +41,18 @@ func (it *instructionTranslator) setMetadata(is *instructionSet, set *bytecode.I
 	}
 }
 
-func (it *instructionTranslator) parseBooleanParam(param string) bool {
-	boolValue, err := strconv.ParseBool(param)
-
-	// Can happen only in case of programmatic error, as the `param` value
-	// is the string version of a boolean.
-	if err != nil {
-		panic(fmt.Sprintf("Unknown boolean value: %s", param))
-	}
-
-	return boolValue
-}
-
-func (it *instructionTranslator) parseParam(param string) interface{} {
-	integer, e := strconv.ParseInt(param, 0, 64)
-	if e != nil {
-		return param
-	}
-
-	i := int(integer)
-
-	return i
-}
-
 func (it *instructionTranslator) transferInstructionSets(sets []*bytecode.InstructionSet) []*instructionSet {
 	iss := []*instructionSet{}
```

Comment:
Can save a few allocations here by doing make([]*instructionSeet, 0, len(sets)). Also, do the instructionSets need to be pointers? Mind you, that each one will be a separate allocation.

Reply:
Anyway, this is actually redundant so removing it reduced 1 additional allocation 😄